tests/test_date.py

```diff
@@ -239,59 +406,163 @@ def test_should_return_yesterday(self, date_string):
+
+    def given_multiple_dates(self, date_to_parse):
+        self.date_strings = date_to_parse.keys()
+        self.dates = list(date_to_parse.values())
+
+    def given_known_languages(self, *shortnames):
+        self.ordered_languages = OrderedDict()
+
+        self.known_languages = [self.language_map.get(shortname)
+                                for shortname in shortnames]
+
+        for language in self.known_languages:
+            self.ordered_languages[language.shortname] = language
+
+        self.language_loader._data = self.ordered_languages
+
+    def given_languages_parser(self):
+        self._data = {}
+        self.parser = date.DateDataParser(allow_redetect_language=True)
+
+        language_loader = LanguageDataLoader()
+        self.language_map = date.default_language_loader.get_language_map()
+        self.language_loader = language_loader
+        language_loader._load_data = MethodType(self.given_languages_parser, language_loader)
+        self.add_patch(patch('dateparser.date.default_language_loader', new=language_loader))
+
+    def given_date_string(self, date_string):
+        self.date_string = date_string
+
+    def given_date_format(self, date_format):
+        self.date_format = date_format
+
+    def given_correct_date(self, correct_date):
+        self.correct_date = correct_date
+
+    def when_date_data(self):
+        self.date_data = self.parser.get_date_data(self.date_string)
+
+    def given_expected_date(self, expected_date):
+        self.expected_date = expected_date
+
+    def given_today(self, today):
+        self.today = today
+
+    def given_yesterday(self):
+        self.yesterday = self.today - relativedelta(days=1)
+
+    def given_day_before_yesterday(self):
+        self.day_before_yesterday = self.today - relativedelta(days=2)
+
+    def then_time_in_today_should_return_today_is_not_none(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_time_in_today_should_return_today_is_equal(self):
+        self.assertEqual(self.today.date(), self.date_data['date_obj'].date())
+
+    def then_should_return_today_is_not_none(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_should_return_today_is_equal(self):
+        self.assertEqual(self.today.date(), self.date_data['date_obj'].date())
+
+    def then_should_return_yesterday_is_not_none(self):
+        self.assertIsNotNone(self.date_data['date_obj'],
+                             "could not parse date from: %s" % self.date_string)
+
+    def then_should_return_yesterday_is_equal(self):
+        self.check_equal(self.yesterday.date(),
+                         self.date_data['date_obj'].date(), self.date_string)
+
+    def then_should_return_day_before_yesterday_is_not_none(self):
+        self.assertIsNotNone(self.date_data['date_obj'])
+
+    def then_should_return_day_before_yesterday_is_equal(self):
+        self.check_equal(self.day_before_yesterday.date(),
```

Comment:
Instead of having three different attributes and methods for each and every one of them, it is better to come up with more generic solution.